cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -311,11 +319,17 @@ public ArtifactDetail getArtifact(Id.Artifact artifactId) throws IOException, Ar
    * @throws ArtifactAlreadyExistsException if the artifact already exists
    * @throws InvalidArtifactException if the artifact is invalid. For example, if it is not a zip file,
    *                                  or the application class given is not an Application.
+   * @throws UnauthorizedException if the current user does not have the privilege to add an artifact in the specified
+   *                               namespace. To add an artifact, a user needs {@link Action#WRITE} privilege on the
+   *                               namespace in which the artifact is being added. If authorization is successful, and
+   *                               the artifact is added successfully, then the user gets {@link Action#ALL} privileges
+   *                               on the added artifact.
    */
-  public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile)
-    throws IOException, WriteConflictException, ArtifactAlreadyExistsException, InvalidArtifactException {
-
+  public ArtifactDetail addArtifact(Id.Artifact artifactId, File artifactFile) throws Exception {
```

Comment:
its a lot of exceptions, but I wouldn't consolidate them.  Otherwise callers have to rely on the javadoc to know what is thrown